### ISSUE TITLE
More JDK Nullable annotations

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#385]
```
-Xlint helped me find Nullable annotations I had previously missed on return 
types:

package java.lang;

public class ClassLoader {
    public @Nullable InputStream getResourceAsStream(String name);
}

package java.lang.management;

public interface ThreadMXBean {
    public @Nullable ThreadInfo getThreadInfo(long id, int maxDepth);
}

package java.lang.reflect;

public class Method {
    public @Nullable Object invoke(@Nullable Object obj, @Nullable Object... args);
}

public class InvocationTargetException {
    public @Nullable Throwable getCause();
}

package java.nio.channels;

public class FileChannel {
    public @Nullable FileLock tryLock();
}

package java.util.concurrent;

public class ConcurrentMap<K, V> {
    @Nullable V putIfAbsent(K key, V value);
}

package java.util.jar;

public class Attributes {
    public @Nullable String getValue(String name);
}

public class JarInputStream {
    public @Nullable Manifest getManifest();
    public @Nullable JarEntry getNextJarEntry();
}

package javax.management.openmbean;

public class CompositeType {
    public @Nullable OpenType<?> getType(String itemName);
}

package java.lang.instrument;

public interface Instrumentation {
    Class[] getInitiatedClasses(@Nullable ClassLoader loader);
}

```

Original issue reported on code.google.com by `trask.st...@gmail.com` on 24 Dec 2014 at 3:43

**Comment by arnehormann** [comment]
For Method.invoke, you also need to make the first argument nullable. That's required to [call static methods](https://docs.oracle.com/javase/7/docs/api/java/lang/reflect/Method.html#invoke(java.lang.Object,%20java.lang.Object...)) - it says so in the second paragraph.

**Comment by mernst** [comment]
This comment in file checker-framework/checker/jdk/nullness/src/java/lang/reflect/Method.java explains why the first argument is not marked as `@Nullable':

```
    // The method being invoked might be one that requires non-null
    // arguments, or might be one that permits null.  We don't know which.
    // Therefore, the Nullness Checker should conservatively issue a
    // warning whenever null is passed, in order to give a guarantee that
    // no nullness-related exception will be thrown by the invoked method.
    public @Nullable Object invoke(Object obj, Object ... args)
```

**Comment by arnehormann** [comment]
Thanks @mernst , I read that beforehand.
Sorry, I read "requires non-null arguments" as addressing the varargs. I know the callee is the first argument in the JVM from ASM. But I didn't identify it as the first argument here.

I just want to get rid of the warning for a reflection based call to a static method.

Soooo, is there a way to skip a check? I didn't find any in the docs and rather doubt it, but right now this messes up my build process... and tests, as I set up Gradle as my testrunner in IntelliJ and it halts on a warning.

**Comment by mernst** [comment]
Arne-

Yes, you can suppress the warning when you know information that is beyond the capabilities of the type system (which sounds like it is true in this case).  Information about suppressing warnings appears at http://types.cs.washington.edu/checker-framework/current/checker-framework-manual.html#suppressing-warnings .  However, let us know if you have questions,

**Comment by mernst** [comment]
Also, thanks for pointing out that the comment was ambiguous.  I updated it in commit ea4ecb7878cbea66a12165b305f49c331e87f886.

**Comment by mernst** [comment]
Closed by 68a87fc
